test/integration/component/test_multiple_public_interfaces.py

```diff
@@ -0,0 +1,1365 @@
+            isinstance(vm_response, list),
+            True,
+            "Check list VM returns a valid list"
+        )
+
+        self.assertNotEqual(
+            len(vm_response),
+            0,
+            "Check Port Forwarding Rule is created"
+        )
+        self.assertEqual(
+            vm_response[0].state,
+            'Running',
+            "VM state should be Running before creating a NAT rule."
+        )
+
+        # Open up firewall port for SSH
+        firewall_rule = FireWallRule.create(
+            self.apiclient,
+            ipaddressid=ip_address.ipaddress.id,
+            protocol=self.services["natrule"]["protocol"],
+            cidrlist=['0.0.0.0/0'],
+            startport=self.services["natrule"]["publicport"],
+            endport=self.services["natrule"]["publicport"]
+        )
+
+        # Get the router details associated with account
+        routers = list_routers(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid,
+        )
+        router = routers[0]
+
+        if (self.hypervisor.lower() == 'vmware'
+                or self.hypervisor.lower() == 'hyperv'):
+            result = get_process_status(
+                self.apiclient.connection.mgtSvr,
+                22,
+                self.apiclient.connection.user,
+                self.apiclient.connection.passwd,
+                router.linklocalip,
+                'iptables -t filter -L FORWARD  -v',
+                hypervisor=self.hypervisor
+            )
+        else:
+            hosts = list_hosts(
+                self.apiclient,
+                id=router.hostid,
+            )
+            self.assertEqual(
+                isinstance(hosts, list),
+                True,
+                "Check for list hosts response return valid data"
+            )
+            host = hosts[0]
+            host.user = self.services["configurableData"]["host"]["username"]
+            host.passwd = self.services["configurableData"]["host"]["password"]
+            try:
+                result = get_process_status(
+                    host.ipaddress,
+                    22,
+                    host.user,
+                    host.passwd,
+                    router.linklocalip,
+                    'iptables -t filter -L FORWARD  -v'
+                )
+            except KeyError:
+                self.skipTest(
+                    "Provide a marvin config file with host\
+                            credentials to run %s" %
+                    self._testMethodName)
+
+        self.debug("iptables -t filter -L FORWARD  -v: %s" % result)
+        res = str(result)
+        self.assertEqual(
+            res.count("eth3   eth0    anywhere             anywhere             state RELATED,ESTABLISHED"),
+            1,
+            "Check to ensure there is a iptable rule to accept the RELATED,ESTABLISHED traffic"
+        )
+
+
+        firewall_rule.delete(self.apiclient)
+
```

Comment:
I think creating and validating a firewall rule can be moved to net utility class, as this has possibilities to be reused.